samples/hosting/generic-multi-hosting/Core_7/Sample.MultiHosting/Program.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using NServiceBus;
+
+class Program
+{
+    static async Task Main(string[] args)
+    {
+        Console.Title = "Samples.MultiHosting";
+        #region multi-hosting
+
+        var endpointOneBuilder = ConfigureEndpointOne(Host.CreateDefaultBuilder(args)).Build();
+        var endpointTwoBuilder = ConfigureEndpointTwo(Host.CreateDefaultBuilder(args)).Build();
+        
+        var endpointOneTask = endpointOneBuilder.RunAsync();
+        var endpointTwoTask = endpointTwoBuilder.RunAsync();
+        
+        await Task.WhenAll(endpointOneTask, endpointTwoTask);
+
+        #endregion
+    }
+    
+    private static IHostBuilder ConfigureEndpointOne(IHostBuilder builder)
+    {
+        builder.UseConsoleLifetime();
+        builder.ConfigureLogging((ctx, logging) =>
```

Comment:
@danielmarbach this one got me thinking, given that NSB uses static logging won't its be broken since both instances will essentially share loggers?

Reply:
Yep last one wins. So as long as the logging configuration is the same it will work but there is no way to differentiate the endpoints in the log other than adding some contextual information to the logger. I think the logger has a dedicated way to [scope things](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.beginscope) but I haven't looked closer into this.

Another option would be that we decorate things internally somehow but I would have to think about it

Reply:
Should we raise an issue for this? or is there anything for it in our backlog (couldn't find any open issue).

Reply:
Raised https://github.com/Particular/PlatformArchitecture/pull/578